apps/ui/lib/api/utils/decodeSystemMessage.mjs

```diff
@@ -1,14 +1,12 @@
-import { BOT_COMMAND, BOT_COMMAND_UI } from '../../system/index.mjs';
-
-function isPhoneNumber(text) {
-    return /^\+(380[0-9]{9}|7[0-9]{10})$/.test(text);
-}
+import { BOT_COMMAND, BOT_COMMAND_UI, COUNTRIES } from '../../system/index.mjs';
+import { Phone } from '../../models/index.mjs';
 
 export function getCommand(message) {
     const { text, data = '', new_chat_member = {} } = message;
+
     if (new_chat_member.status === 'kicked') return BOT_COMMAND.LEAVE_CHAT;
     if (new_chat_member.status === 'member') return BOT_COMMAND.RESTART_CHAT;
-    if (isPhoneNumber(text)) return BOT_COMMAND.ADD_TRACK_PHONE;
+    if (Phone.isValid(text, COUNTRIES)) return BOT_COMMAND.ADD_TRACK_PHONE;
```

Comment:
in this place could be a function that identifies that this is a phone as was prev. but based on another rule.
Phone validation should be in the use-case.